tests/pr_test.go

```diff
@@ -2,22 +2,35 @@
 package test
 
 import (
+	"os"
 	"testing"
 
 	"github.com/stretchr/testify/assert"
+	"github.com/terraform-ibm-modules/ibmcloud-terratest-wrapper/cloudinfo"
 	"github.com/terraform-ibm-modules/ibmcloud-terratest-wrapper/testhelper"
 )
 
 // Use existing resource group
 const resourceGroup = "geretain-test-resources"
 const defaultExampleTerraformDir = "examples/basic"
 
+var sharedInfoSvc *cloudinfo.CloudInfoService
+
+// TestMain will be run before any parallel tests, used to set up a shared InfoService object to track region usage
+// for multiple tests
+func TestMain(m *testing.M) {
+	sharedInfoSvc, _ = cloudinfo.NewCloudInfoServiceFromEnv("TF_VAR_ibmcloud_api_key", cloudinfo.CloudInfoServiceOptions{})
+
+	os.Exit(m.Run())
+}
+
 func setupOptions(t *testing.T, prefix string) *testhelper.TestOptions {
 	options := testhelper.TestOptionsDefault(&testhelper.TestOptions{
 		Testing:            t,
 		TerraformDir:       defaultExampleTerraformDir,
 		Prefix:             prefix,
 		ResourceGroup:      resourceGroup,
+		CloudInfoService:   sharedInfoSvc,
```

Comment:
(this comment should really be on line 41, but since there were no changes on that line I'm putting it here)

For PowerSystems, the region selection is not handled in TestOptionsDefault(), so you will need to pass the `sharedInfoSvc` pointer into one more place.

A few lines down from this (line 41) is the `testhelper.GetBestPowerSystemsRegion()` call, this will also need to change. You will need to change to the overloaded method to accept service options (which is same function name but ends in a capital O) and provide the sharedInfoSvc pointer into that options struct.

It should end up looking like this:
```go
options.Region, _ = testhelper.GetBestPowerSystemsRegionO(options.RequiredEnvironmentVars["TF_VAR_ibmcloud_api_key"], options.BestRegionYAMLPath, options.DefaultRegion, &testhelper.TesthelperTerraformOptions{
			CloudInfoService: sharedInfoSvc})
```
This will then use the same `sharedInfoSvc` object for all parallel calls and put each test on a different region.

Reply:
Adding service option(`&testhelper.TesthelperTerraformOptions`) in the overloaded method(`testhelper.GetBestPowerSystemsRegionO`) is giving following error - 

```
./pr_test.go:42:171: cannot use &testhelper.TesthelperTerraformOptions{…} (value of type *testhelper.TesthelperTerraformOptions) as testhelper.TesthelperTerraformOptions value in argument to testhelper.GetBestPowerSystemsRegionO
FAIL    github.com/terraform-ibm-modules/terraform-ibm-powervs-infrastructure [build failed]
make: *** [run-tests-local] Error 1
```

Reply:
Yes you are correct, sorry about that, the options struct is not a pointer, so you were correct to remove the "&".

In the logs I see that this is actually working:
```
2023/02/14 10:15:30 --- new best region is syd04
2023/02/14 10:15:30 Best region was found!: syd04
```
and a little later 
```
2023/02/14 10:15:33 --- new best region is syd05
2023/02/14 10:15:33 Best region was found!: syd05
```

I think the problem is that you are not using that region in your `options.TerraformVars` so your test is always using the default for the example. Line 56 needs to be uncommented and replace the value with `options.Region`.